validator/src/journal/block_manager.rs

```diff
@@ -557,20 +557,59 @@ impl Iterator for GetBlockIterator {
 
 pub struct BranchIterator {
     state: Arc<RwLock<BlockManagerState>>,
+    initial_block_id: String,
     next_block_id: String,
     blockstore: Option<String>,
 }
 
 impl BranchIterator {
     fn new(state: Arc<RwLock<BlockManagerState>>, first_block_id: String) -> Self {
+        let next_block_id = {
+            let mut block_manager = state
+                .write()
+                .expect("No log holder should have poisoned the lock");
```

Comment:
The argument should state the error that occurred, not the expected behavior. Also there is a typo.